modules/registrars/openprovider/OpenProvider/API/ApiV1.php

```diff
@@ -121,14 +123,67 @@ public function call(string $cmd, array $args = []): ResponseInterface
             return $response;
         }
 
+
+        if (isset($reply['warnings'])) {
+            $this->add_warning($reply['warnings'], $cmd, $args);
+        }
+
         $data = $this->serializer->normalize($reply->getData());
         $response = $this->successResponse($response, $data);
-
         $this->log($cmd, $args, $response);
 
         return $response;
     }
 
+    /**
+     * @param array $warnings
+     * @param string $cmd
+     */
+    private function add_warning($warnings, $cmd, $args)
+    {
+        $title = "Warning: ";
+        $description = "";
+        $index = 1;
+        foreach ($warnings as $warn) {
+            if ($warn['code'] != 0) {
+                $description .= "Warning " . $index . ":" . "\nCode: " . $warn["code"] . "\nDescription: " . $warn["desc"]  . "\nData: " . $warn["data"] . "\n";
+            }
+        }
+        $cmd = preg_replace('/(?<!\s)([A-Z])/', ' $1', $cmd); // Insert space before capital letters
+        $cmd = ucwords($cmd); // Capitalize words
+        $title = $title . $cmd . " Command.";
+        if (isset($args['domain']['name']) && isset($args['domain']['extension'])) {
+            $domainName = $args['domain']['name'];
+            $extension = $args['domain']['extension'];
+            $title = $title . " Domain: " . $domainName . "." . $extension;
+        }
+        if (!empty($description) && !empty($title)) {
+            $this->add_todo($title, $description);
+        }
+    }
+
+    /**
+     * Create New To-do item
+     * @param $title
+     * @param $description
+     */
+    private function add_todo($title, $description)
+    {
+        $today = Carbon::now();
+        $tomorrow = $today->addDay();
+
+        $todo = [
+            'date' => $today,
+            'title' => $title,
+            'description' => $description,
+            'admin' => 0,
+            'status' => 'Pending',
```

Comment:
Check the status defined somewhere like model class or better to define in the class at-least

Reply:
As discussed, kept the status defined in the same code line